KIPs/kip-113.md

```diff
@@ -0,0 +1,252 @@
+---
+kip: 113
+title: BLS public key registry
+author: Ian (@ian0371), Ollie (@blukat29), Joseph(@jiseongnoh), Aidan(@aidan-kwon)
+discussions-to: https://github.com/klaytn/kips/issues/113
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-05-22
+---
+
+## Simple Summary
+
+A design of simple BLS12-381 registry.
+
+## Abstract
+
+This standard defines an interface for BLS12-381 registry, which enables nodes or smart contracts to access and view all the public keys of GCs. This is not to replace AddressBook, but rather to supplement it in order to facilitate the use of BLS in various components, including signature-based random (KIP-114).
+
+## Motivation
+
+The current design of AddressBook cannot incorporate BLS12-381 keys. The lack of BLS registry limits the use of BLS from the node. The objective is to design a simple registry that includes essential functions, and minimal access control, thereby reducing any potential security risks.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+Registry contract must implement the IKIP_113 interface:
+
+```solidity
+pragma solidity ^0.8.19;
+
+/// @title KIP-113 BLS public key registry
+/// @dev See https://github.com/klaytn/kips/issues/113
+interface IKIP_113 {
```

Comment:
Comparing to other implementations, IKIP113 is more commonly used until now rather than IKIP_113.

```suggestion
interface IKIP113 {
```


https://github.com/klaytn/klaytn-contracts/blob/master/contracts/KIP/interfaces/IKIP13.sol